kore/src/Kore/Builtin/AssociativeCommutative.hs

```diff
@@ -1197,13 +1167,23 @@ unifyEqualsElementLists
             "Duplicated element in unification results"
             first
             second
-        Just remainderTerm -> do
-            opaqueUnifier <- unifyEqualsChildren opaque remainderTerm
-            let (opaqueTerm, opaquePredicate) = Pattern.splitTerm opaqueUnifier
-
-                result = unifier `andCondition` opaquePredicate
+        Just remainderTerm -> case opaque of
+            Var_ _ -> do
+                opaqueUnifier <- unifyEqualsChildren opaque remainderTerm
+                let
+                    (opaqueTerm, opaquePredicate) =
+                        Pattern.splitTerm opaqueUnifier
+                    result = unifier `andCondition` opaquePredicate
+
+                return (result, [opaqueTerm])
+            _ -> (error . unlines)
```

Comment:
If I understand correctly, this throws an error if an opaque collection isn't a variable?

Reply:
Yes. The check I removed (as you pointed out) threw an error for `Application` patterns, but whitelisting seems safer then blacklisting.